source/contributing/implementation/datamanipulator.rst

```diff
@@ -0,0 +1,148 @@
+
+Upon completion of the first steps you'll have to get this done too:
+
+#. Register the ``Key`` in the ``KeyRegistry``
+#. Implement the ``DataProcessor``
+#. Implement the ``ValueProcessor`` for each value being represented by the ``DataManipulator``
+
+1. Implementing the DataManipulator
+===================================
+
+Implementing the ``DataManipulator`` is done by extending ``AbstractData`` in the easiest case. Let's assume we have a
+single value based manipulator, like for example the ``flying`` state on an entity which can be ``true`` or ``false``.
+We decide to abstract things, so now we're able to just extend ``AbstractBooleanData``.
+Heres a short example:
+
+.. code-block:: java
+
+  public class SpongeFlyingData extends AbstractBooleanData<FlyingData, ImmutableFlyingData> implements FlyingData {
+ /* your code here */
+  }
+
+The next step is to provide the generic ``DataManipulator`` that follows the generic of ``FlyingData``. In this case we
+use ``FlyingData.class`` in our ``super`` constructor, followed by the ``Key`` which identifies the recommended
+``value`` of the flying state and last but not least the ``Key`` itself: ``Keys.IS_FLYING``.
+
+.. code-block:: java
+
+ public SpongeFlyingData(boolean flying) {
+          super(FlyingData.class, flying, Keys.IS_FLYING);
+ }
+
+In most cases while implementing an abstract Manipulator you want to have two constructors:
+
+* One without arguments (no-args) which calls the second
+* and one single-argument constructor which is actually used for the provided ``value``
+
+Now, after the Manipulator is done you have to check if you are extending ``AbstractData`` or not.
+If you do, you must call ``registerFieldGetter()`` and ``registerFieldSetter`` in your constructor.
+Don't forget the ``Field`` registration!
+
+.. code-block:: java
+
+ private void registerStuff() {
+      registerFieldGetter(Keys.IS_FLYING, new GetterFunction<Object>() {
+             @Override
+             public Value get() {
+                 return getflying();
+             }
+         });
+      registerFieldSetter(Keys.IS_FLYING, new SetterFunction<Object>() {
+              @Override
+              public void set(Object value) {
+                   setFlying(((Boolean) value).booleanValue());
+              }
+          });
+  }
+
+That's it. The ``DataManipulator`` should be done now.
+
+2. ImmutableDataManipulator
+===========================
+
+Just repeat the above mentioned steps for the ``ImmutableDataManipulator``.
+
+Now, after the ``Manipulators`` are done we have to register the ``KEY`` and finally implement the
+``DataProcessor`` and ``ValueProcessor``.
+
+3. Register the Key in the KeyRegistry
+======================================
+
+The next step is to register your ``KEYS`` in the ``KeyRegistry``:
+
+.. code-block:: java
+
+ public static void registerKeys() {
+      keyMap.put("is_flying", makeSingleKey(Boolean.class, Value.class, of("IsFlying")));
```

Comment:
The first string is exactly as you assumed, it's to match the field name in `Keys`. The second is the `DataQuery` suggested for use throughout setting/getting that information from a `DataView`. It's not necessarily matching Minecraft's tag id's.